topdown/http_test.go

```diff
@@ -2514,3 +2518,99 @@ var httpSendHelperRules = []string{
 		no_headers = object.remove(resp, ["headers"])
 	}`,
 }
+
+func httpSocket(t *testing.T, path string) (net.Listener, error) {
+	t.Helper()
+	sock, err := net.Listen("unix", path)
+	if err != nil {
+		return nil, err
+	}
+
+	// install signal handler to clean up
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
+	go func(c chan os.Signal) {
+		<-c
+		t.Cleanup(func() { sock.Close() })
+	}(sigc)
```

Comment:
Do we need the signal handler? AFAIK, any function registered with `t.Cleanup()` will be called when it's time to clean up. 

I guess we could make `httpSocket` return a function for cleanup, like
```go
socket, cleanup, err := httpSocket(t, socketPath)
if err != nil { /* ... */ }
t.Cleanup(cleanup)
```

That said, what will this give us over that `rs.Close()` takes care of anyways? [The docs](https://pkg.go.dev/net/http#Server.Close) make the think it'll close the listener 🤔

Reply:
I will take a look at removing the signal handler